tests: spi_loopback: Fix same buf cmd test

decsny opened the pull request:
The tests was written wrong. It was meant to test using same spi bufs
for both rx and tx, as in tree many sensor and other spi device drivers
use this paradigm. But the 2nd buf setup call was overwriting the first.
Fix by not using the helper function for this case.

And for the write back test, test using same spi_buf_set.

Fixes #91384

natto1784 commented:
I have been failing to understand some things here - Whether you overwrite or not, the tx.buffers and rx.buffers in the end still remain the same since you use `rx_bufs_pool` for both.

Which makes the new same_buf_cmd test essentially the same as
```
	const struct spi_buf_set tx = spi_loopback_setup_xfer(tx_bufs_pool, 2,
							      buffer_rx, 1,
							      buffer_rx+1, BUF_SIZE - 1);
	const struct spi_buf_set rx = spi_loopback_setup_xfer(rx_bufs_pool, 2,
							      buffer_rx, 1,
							      buffer_rx+1, BUF_SIZE - 1);
```

Please correct me if I am wrong

Commit:
tests: spi_loopback: Fix same_buf_cmd test

The tests was written wrong. It was meant to test using same spi bufs
for both rx and tx, as in tree many sensor and other spi device drivers
use this paradigm. But the 2nd buf setup call was overwriting the first.
Fix by not using the helper function for this case.

And for the write back test, test using same spi_buf_set.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

decsny commented:
@tbursztyka please re-review

decsny commented:
@tbursztyka i added myself as assignee for now bc you approved already, and it should be pretty uncontroversial to fix this test case, hope its fine

danieldegrasse commented:
Good with this given that @tbursztyka previously approved.
